versioned/tiered/tiered-tests/src/main/java/com/dremio/nessie/versioned/impl/AbstractTestStore.java

```diff
@@ -355,6 +355,86 @@ void putWithConditionL3() {
     putWithCondition(ValueType.L3, SampleEntities.createL3(random));
   }
 
+  @Test
+  void deleteWithConditionValue() {
+    deleteWithCondition(ValueType.VALUE, SampleEntities.createValue(random));
+  }
+
+  @Test
+  void deleteWithConditionBranch() {
+    deleteWithCondition(ValueType.REF, SampleEntities.createBranch(random));
+  }
+
+  @Test
+  void deleteWithConditionTag() {
+    deleteWithCondition(ValueType.REF, SampleEntities.createTag(random));
+  }
+
+  @Test
+  void deleteWithConditionCommitMetadata() {
+    deleteWithCondition(ValueType.COMMIT_METADATA, SampleEntities.createCommitMetadata(random));
+  }
+
+  @Test
+  void deleteWithConditionKeyFragment() {
+    deleteWithCondition(ValueType.KEY_FRAGMENT, SampleEntities.createFragment(random));
+  }
+
+  @Test
+  void deleteWithConditionL1() {
+    deleteWithCondition(ValueType.L1, SampleEntities.createL1(random));
+  }
+
+  @Test
+  void deleteWithConditionL2() {
+    deleteWithCondition(ValueType.L2, SampleEntities.createL2(random));
+  }
+
+  @Test
+  void deleteWithConditionL3() {
+    deleteWithCondition(ValueType.L3, SampleEntities.createL3(random));
+  }
+
+  @Test
+  void deleteValue() {
+    delete(ValueType.VALUE, SampleEntities.createValue(random));
+  }
+
+  @Test
+  void deleteBranch() {
+    delete(ValueType.REF, SampleEntities.createBranch(random));
+  }
+
+  @Test
+  void deleteTag() {
+    delete(ValueType.REF, SampleEntities.createTag(random));
+  }
+
+  @Test
+  void deleteCommitMetadata() {
+    delete(ValueType.COMMIT_METADATA, SampleEntities.createCommitMetadata(random));
+  }
+
+  @Test
+  void deleteKeyFragment() {
+    delete(ValueType.KEY_FRAGMENT, SampleEntities.createFragment(random));
+  }
+
+  @Test
+  void deleteL1() {
+    delete(ValueType.L1, SampleEntities.createL1(random));
+  }
+
+  @Test
+  void deleteL2() {
+    delete(ValueType.L2, SampleEntities.createL2(random));
+  }
+
+  @Test
+  void deleteL3() {
+    delete(ValueType.L3, SampleEntities.createL3(random));
+  }
+
```

Comment:
Did you investigate any multithreaded tests?

Reply:
I have not gotten the chance to do that. I will be spending an hour tomorrow tomorrow morning to investigate multithreaded tests.

Reply:
I put a TODO into the delete() implementation to note the possible locking of the critical section within the transaction.

Reply:
Noted